manifests/pip.pp

```diff
@@ -221,14 +219,17 @@
 
       default: {
         # Anti-action, uninstall.
-        $pip_exec_name = "pip_uninstall_${name}"
+        $uninstall = true
         $command = "echo y | ${pip_env} uninstall ${uninstall_args} ${proxy_flag} ${name}"
         $unless_command = "! ${pip_env} list | grep -i -e '${grep_regex}'"
       }
     }
   }
 
-  exec { $pip_exec_name:
+  exec { (defined('$uninstall')) ? {
+      true  => "pip_uninstall_${name}",
+      false => "pip_install_${name}",
+    }:
```

Comment:
Can you please do this outside of the resource title? This will help to make it better readable.